TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/LatchSystem.java

```diff
@@ -2,47 +2,79 @@
 
 import com.qualcomm.robotcore.hardware.Servo;
 
+import java.util.EnumMap;
+
 
 public class LatchSystem {
-    private Servo servoLeft;
-    private Servo servoRight;
-    public boolean latched;
-    private final double LEFT_DOWN_POSITION = 0.714623491755917;
-    private final double RIGHT_DOWN_POSITION = 0.18877972287358707;
-    private final double LEFT_UP_POSITION = 0.4461410963209186;
-    private final double RIGHT_UP_POSITION = 0.45632352852029817;
-
-    public LatchSystem(Servo left, Servo right) {
-        this.servoLeft = left;
-        this.servoRight = right;
-        initServo();
+
+    public enum Latch {
+        // i don't think I'm allowed to but a boolean thing here but who knows!
+        LEFT (0.715, 0.446, false),
+        RIGHT (0.189, 0.456, false);
+
+        private final double upPosition;
+        private final double downPosition;
+        private boolean latched;
+
+        Latch(double downPosition, double upPosition, boolean latched) {
+            this.downPosition = downPosition;
+            this.upPosition = upPosition;
+            this.latched = latched;
+        }
+
+        private double upPosition() {
+            return upPosition;
+        }
+
+        private double downPosition() {
+            return downPosition;
+        }
+
+        private boolean getLatched() {
+            return latched;
+        }
+
+        private void setLatched(boolean isLatched) {
+            latched = isLatched;
+        }
     }
 
+    public EnumMap<Latch, Servo> latches;
+
+
     private void initServo() {
-        unlatch();
+        latches.forEach((name, servo) -> {
+            servo.setPosition(name.upPosition());
+            name.latched = false;
```

Comment:
if `name.upPosition` is set at this point, presumably `name.latched` is as well.